src/context/TreeContext.tsx

```diff
@@ -25,7 +21,28 @@ TreeContext.displayName = "TreeContext";
 export default function TreeContextProvider({ children }: AppProps) {
   const [tree, setTree] = React.useState<TreeNode[]>([]);
 
-  const value = { tree };
+  // Inserts a node to the correct parent
+  const insertNode = (newNode: TreeNode) => {
+    // If the current state of the tree is empty, then it will be the root node
+    if (tree.length === 0) {
+      setTree((oldTree) => [...oldTree, newNode]);
+    } else {
+      // If there is already a root node, then we add children to the correct parent
+      let parentName = newNode.parent;
+      const rootNode = tree[0];
+
+      // If the parent is null (user did not specify), the parent will be the root node
+      if (parentName === null) {
+        newNode.parent = rootNode.name;
+        parentName = newNode.parent;
+      }
+      // Helper function to find the correct parent and insert the new child
+      const newTree = findNodeAndInsert(parentName, newNode, rootNode);
```

Comment:
Can we pass parameters as an object because people may mess the order of these parameters?

`findNodeAndInsert({parentName, newNode, rootNode});`

Reply:
Yes and everything seems to be working fine. I made an interface for the object passed to this function, so the function itself isn't cluttered with type declarations.